TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/OrderedStreamLifecyleManager.java

```diff
@@ -0,0 +1,17 @@
+package org.opensearch.migrations.trafficcapture;
+
+import java.util.concurrent.CompletableFuture;
+
+public abstract class OrderedStreamLifecyleManager extends StreamLifecycleManager {
+    CompletableFuture<Object> futureForLastClose = CompletableFuture.completedFuture(null);
+
+    protected abstract CodedOutputStreamHolder createStream();
+
+    protected CompletableFuture<Object> closeStream(CodedOutputStreamHolder outputStreamHolder, int index) {
+        futureForLastClose = futureForLastClose.thenCompose(v -> kickoffCloseStream(outputStreamHolder, index));
+        return futureForLastClose;
+    }
+
+    protected abstract CompletableFuture<Object> kickoffCloseStream(CodedOutputStreamHolder outputStreamHolder,
```

Comment:
Reading this at face value I don't know when I should closeStream or kickoffCloseStream. I initially thought I might have to call kickoffCloseStream before I called closeStream. Is it safe to say this is `closeAllStreamsAsync` and `closeStreamAsync`? Not tied to that name but feel like we need something to make this clearer

Reply:
I was not sure about the nomenclature that I chose at the time & I'm less sure of it now.
closeStream is inherited from StreamLifecycleManager.  That's what callers will be using.  The OrderedStreamLifecycleManager implements a version of closeStream() that delegates to the subclass to create a CompletableFuture and kick if off.  Nobody other than the superclass should likely call kickoffCloseStream since that will just return one CompletableFuture, not a CompletableFuture with all of the work chained together.